Source/WebCore/svg/SVGLengthContext.cpp

```diff
@@ -268,14 +268,37 @@ ExceptionOr<float> SVGLengthContext::convertValueFromUserUnits(float value, SVGL
 
 float SVGLengthContext::computeNonCalcLength(float inputValue, CSS::LengthUnit unit) const
 {
-    if (conversionToCanonicalUnitRequiresConversionData(unit)) {
-        auto conversionData = cssConversionData();
-        if (!conversionData)
-            return 0.0f;
-        return clampTo<float>(Style::computeNonCalcLengthDouble(inputValue, unit, *conversionData));
+    if (!conversionToCanonicalUnitRequiresConversionData(unit))
+        return clampTo<float>(Style::computeNonCalcLengthDouble(inputValue, unit, { }));
+
+
+    auto conversionData = cssConversionData();
+    if (!conversionData)
+        return 0.0f;
+
+    auto resolvedValue = clampTo<float>(Style::computeNonCalcLengthDouble(inputValue, unit, *conversionData));
+
+    if (unit == CSS::LengthUnit::Em || unit == CSS::LengthUnit::Rem) {
```

Comment:
Can we have test cases for other font units as well?

It would also good to add a comment why the other font units don't have special logic (maybe FIXME).

Reply:
added comment